infrastructure-playbooks/cephadm-adopt.yml

```diff
@@ -0,0 +1,689 @@
+---
+#
+# This playbook does a cephadm adopt for all the Ceph services
+#
+
+- name: confirm whether user really meant to adopt the cluster by cephadm
+  hosts: localhost
+  connection: local
+  become: false
+  gather_facts: false
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to adopt the cluster by cephadm ?
+      default: 'no'
+      private: no
+  tasks:
+    - name: exit playbook, if user did not mean to adopt the cluster by cephadm
+      fail:
+        msg: >
+          Exiting cephadm-adopt playbook, cluster was NOT adopted.
+           To adopt the cluster, either say 'yes' on the prompt or
+           use `-e ireallymeanit=yes` on the command line when
+           invoking the playbook
+      when: ireallymeanit != 'yes'
+
+- name: gather facts and prepare system for cephadm
+  hosts:
+    - "{{ mon_group_name|default('mons') }}"
+    - "{{ osd_group_name|default('osds') }}"
+    - "{{ mds_group_name|default('mdss') }}"
+    - "{{ rgw_group_name|default('rgws') }}"
+    - "{{ mgr_group_name|default('mgrs') }}"
+    - "{{ rbd_mirror_group_name|default('rbdmirrors') }}"
+    - "{{ nfs_group_name|default('nfss') }}"
+    - "{{ iscsi_gw_group_name|default('iscsigws') }}"
+    - "{{ grafana_server_group_name|default('grafana-server') }}"
+  become: true
+  gather_facts: false
+  vars:
+    delegate_facts_host: true
+  tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - name: fail if one osd node is using filestore
+      fail:
```

Comment:
Also it would be nice to be sure that the cluster is in a good shape.  Checking if we have a "HEALTH_OK" cluster , and inform if there is any health alarm could be useful. The user can abort the adoption or continue with it at his own risk.

Reply:
in fact we're already doing this for the upgrade playbook so I think we can do the same here

https://github.com/ceph/ceph-ansible/blob/master/infrastructure-playbooks/rolling_update.yml#L127-L142